synapse/util/caches/dictionary_cache.py

```diff
@@ -117,7 +247,13 @@ def invalidate(self, key: KT) -> None:
         # Increment the sequence number so that any SELECT statements that
         # raced with the INSERT don't update the cache (SYN-369)
         self.sequence += 1
-        self.cache.pop(key, None)
+
+        # We want to drop all information about the dict for the given key, so
+        # we use `del_multi` to delete it all in one go.
+        #
+        # We ignore the type error here `del_mutli` accepts a truncated key
+        # (when the key type is a tuple).
+        self.cache.del_multi((key,))  # type: ignore[arg-type]
```

Comment:
so how did this ever work before?

Reply:
It wasn't a `TreeCache` before: it was just using a `DictionaryEntry` as the cache value, and keeping track of whether that was "full" or "partial".

Reply:
> It wasn't a TreeCache before: 

oh gosh, I'd completely missed that.